modules/ipi-install-configure-network-components-to-run-on-the-control-plane.adoc

```diff
@@ -0,0 +1,126 @@
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  name: 50-worker-fix-ipi-rwn
+  labels:
+    machineconfiguration.openshift.io/role: worker
+spec:
+  config:
+    ignition:
+      version: 3.1.0
+    systemd:
+      units:
+      - name: nodeip-configuration.service
+        enabled: true
+        contents: |
+          [Unit]
+          Description=Writes IP address configuration so that kubelet and crio services select a valid node IP
+          Wants=network-online.target
+          After=network-online.target ignition-firstboot-complete.service
+          Before=kubelet.service crio.service
+          [Service]
+          Type=oneshot
+          ExecStart=/bin/bash -c "exit 0 "
+          [Install]
+          WantedBy=multi-user.target
+    storage:
+      files:
+        - contents:
+            source: data:,
+            verification: {}
+          filesystem: root
+          mode: 420
+          path: /etc/kubernetes/manifests/keepalived.yaml
+        - contents:
+            source: data:,
+            verification: {}
+          filesystem: root
+          mode: 420
+          path: /etc/kubernetes/manifests/mdns-publisher.yaml
+        - contents:
+            source: data:,
+            verification: {}
+          filesystem: root
+          mode: 420
+          path: /etc/kubernetes/manifests/coredns.yaml
+----
++
+This manifest will place the `apiVIP` and `ingressVIP` virtual IP addresses on the control plane nodes. Additionally, the following processes will deploy on the control plane nodes only:
++
+* `openshift-ingress-operator`
++
+* `keepalived`
+
+. Save the `cluster-network-avoid-workers-99-config.yaml` file.
+
+. Create a `manifests/cluster-ingress-default-ingresscontroller.yaml` file.
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  nodePlacement:
+    nodeSelector:
+      matchLabels:
+        node-role.kubernetes.io/master: ""
+----
+
+. Consider backing up the `manifests` directory. The installer deletes the `manifests/` directory when creating the cluster.
+
+. Modify the `cluster-scheduler-02-config.yml` manifest to make the control plane nodes schedulable by setting the `mastersSchedulable` field to `true`. Control plane nodes are not schedulable by default. For example:
++
+----
+$ sed -i "s;mastersSchedulable: false;mastersSchedulable: true;g" clusterconfigs/manifests/cluster-scheduler-02-config.yml
+----
++
+[NOTE]
+====
+If control plane nodes are not schedulable, deploying the cluster will fail.
+====
+
+. Before deploying the cluster, ensure that the `api.<cluster-name>.<domain>` domain name is resolvable in the DNS. When you configure network components to run exclusively on the control plane, the internal DNS resolution no longer works for worker nodes, which is an expected outcome.
```

Comment:
I think you could use less words in this sentence with no ill effect, e.g., "When you configure network components to run exclusively on the control plane, the internal DNS resolution is disabled for worker nodes."

Reply:
I tried something shorter already, but Andrew Beekhof specifically requested this wording because the internal DNS resolution would normally work otherwise, and he wanted it to be clear that this is an expected outcome.